internal/providers/providers.go

```diff
@@ -78,30 +78,27 @@ func GetCredentialStateForProvider(
 	s db.Store,
 	cryptoEngine crypto.Engine,
 	provCfg *serverconfig.ProviderConfig,
-) string {
-	var credState string
+) (string, error) {
```

Comment:
While fixing up unit tests, I discovered that there is a latent bug involving this function: When it was written originally, it swallowed errors because the one place where it was used didn't care if it failed or not. The function subsequently was reused in another place where the errors were significant, but this function would swallow the errors, leading to unexpected behaviour.

I refactored the function to return errors, and in doing so, made some changes to simplify the control flow of the function.